users/models.py

```diff
@@ -59,6 +64,13 @@ class DataOceanUser(AbstractUser):
     def __str__(self):
         return self.email
 
+    def create_project(self, project_name, project_description=None):
+        new_token = generate_key()
+        new_project = Project.objects.create(name=project_name, token=new_token,
+                                             description=project_description)
+        new_project.users.add(self)
+        return new_project
```

Comment:
now you can do here just 
```suggestion
    def create_project(self, project_name, project_description=None):
        return self.user_projects.create(
            name=project_name,
            token=generate_key(),
            description=project_description,
        )
```

Reply:
If you want, you can create token in `save()` method in Project. It will be nice.

Reply:
Please, explain the reason